iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/GeneratorXml.java

```diff
@@ -63,6 +63,22 @@ protected GeneratorAdder createAdder(VoltageLevel vl) {
 
     @Override
     protected Generator readRootElementAttributes(GeneratorAdder adder, NetworkXmlReaderContext context) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    protected void readElement(String id, GeneratorAdder adder, NetworkXmlReaderContext context) throws XMLStreamException {
+        boolean voltageRegulatorOn = readRootElementAttributesGenerator(adder, context);
+        Generator g = adder.add();
+        readPQ(null, g.getTerminal(), context.getReader());
+        readSubElementsGenerator(g, context);
+        if (voltageRegulatorOn && g.getRegulatingTerminal() == null) {
```

Comment:
This condition should be only if IIDM version < 1.6.
This PR is for me a breaking change in serialization and should justify a new version. Luckily, we bumped in 1.6 so this should be okay to only check that the version of the context is < 1.6